newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -68,81 +83,70 @@ type RootMenuTemplate =
+      this.props.onCloseApp()
     );
-    ipcRenderer.on(
-      ('main-menu-export': MainMenuEvent),
-      event => this._editor && this._editor.openExportDialog()
+    ipcRenderer.on(('main-menu-export': MainMenuEvent), event =>
+      this.props.onExportProject()
     );
-    ipcRenderer.on(
-      ('main-menu-create': MainMenuEvent),
-      event => this._editor && this._editor.openCreateDialog()
+    ipcRenderer.on(('main-menu-create': MainMenuEvent), event =>
+      this.props.onCreateProject()
     );
-    ipcRenderer.on(
-      ('main-menu-open-project-manager': MainMenuEvent),
-      event => this._editor && this._editor.openProjectManager()
+    ipcRenderer.on(('main-menu-open-project-manager': MainMenuEvent), event =>
+      this.props.onOpenProjectManager()
     );
-    ipcRenderer.on(
-      ('main-menu-open-start-page': MainMenuEvent),
-      event => this._editor && this._editor.openStartPage()
+    ipcRenderer.on(('main-menu-open-start-page': MainMenuEvent), event =>
+      this.props.onOpenStartPage()
     );
-    ipcRenderer.on(
-      ('main-menu-open-debugger': MainMenuEvent),
-      event => this._editor && this._editor.openDebugger()
+    ipcRenderer.on(('main-menu-open-debugger': MainMenuEvent), event =>
+      this.props.onOpenDebugger()
     );
-    ipcRenderer.on(
-      ('main-menu-open-about': MainMenuEvent),
-      event => this._editor && this._editor.openAboutDialog()
+    ipcRenderer.on(('main-menu-open-about': MainMenuEvent), event =>
+      this.props.onOpenAbout()
     );
-    ipcRenderer.on(
-      ('main-menu-open-preferences': MainMenuEvent),
-      event => this._editor && this._editor.openPreferences()
+    ipcRenderer.on(('main-menu-open-preferences': MainMenuEvent), event =>
+      this.props.onOpenPreferences()
     );
-    ipcRenderer.on(
-      ('main-menu-open-language': MainMenuEvent),
-      event => this._editor && this._editor.openLanguage()
+    ipcRenderer.on(('main-menu-open-language': MainMenuEvent), event =>
+      this.props.onOpenLanguage()
     );
-    ipcRenderer.on(
-      ('main-menu-open-profile': MainMenuEvent),
-      event => this._editor && this._editor.openProfile()
+    ipcRenderer.on(('main-menu-open-profile': MainMenuEvent), event =>
+      this.props.onOpenProfile()
     );
-    ipcRenderer.on(
-      ('update-status': MainMenuEvent),
-      (event, status) => this._editor && this._editor.setUpdateStatus(status)
+    ipcRenderer.on(('update-status': MainMenuEvent), (event, status) =>
+      this.props.setUpdateStatus(status)
     );
 
     this._buildAndSendMenuTemplate();
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: Props) {
     if (this.props.i18n.language !== this._language) {
       this._buildAndSendMenuTemplate();
       this._language = this.props.i18n.language;
     }
+
+    // Update menu if a project has just been opened or closed
+    if (!!prevProps.project !== !!this.props.project) {
```

Comment:
This won't see the difference between a project and a new project, because you're converting them to booleans which are just true or false.
If a project is changed to another, the object will change, but the boolean will stay true.

Instead, do simply:
`prevProps.project !== this.props.project`

Reply:
Actually, I wanted to update the menu only if the user goes from "no project open" to "some project open" or vice versa, since switching projects does not change if a menu item is enabled or not. 

Although I guess if we have a menu item in the future whose `enabled` status does depend on the project open, rebuilding menu on any project change will be better (after all this probably does not have any performance-related issue), so I guess you're right too. I'll change this.

Reply:
Ah I see - you're right that it worked in this case. But as you're saying in the future enabled or even the menu items could change according to `this.props.project` - so better rebuild the menu in case of change :)